### DISCUSSION TITLE
Using semantic kernel in promptflow

### DISCUSSION THREAD

**Discussion by mylee04** [microsoft/promptflow#2007] [Q&A] [answered]
Hello, I'm trying to use semantic kernel in promptflow. However, I'm facing some problems to utilize it. For example, 

```
INPUTCLASS_PATH = "plugins/WriterPlugin/InputClass" 
kernel = sk.Kernel()
kernel.add_chat_service(
    "chat_completion",
    sk.AzureChatCompletion(
        deployment_name=AZURE_OPEN_AI_CHAT_COMPLETION_DEPLOYMENT_NAME,
        endpoint=AZURE_OPEN_AI_ENDPOINT,
        api_key=AZURE_OPEN_AI_API_KEY,
    ),
)
```
In this case, I don't need a sk.AzureChatCompletion part, I can make it automatically if I use a llm node. So I have no idea how to correctly use semantic kernel in promptflow. Please let me know.

Second question is that using "path". May I upload some plugins(json files and .txt from semantic function) and then call this each folder in promptflow?

Thank you for your help and I really expect any replies on this question.

**Comment by brynn-code** [comment, marked as answer]
Hi, sorry for the late response, please follow this guide on using semantic kernel in promptflow: https://learn.microsoft.com/en-us/semantic-kernel/agents/planners/evaluate-and-deploy-planners/create-a-prompt-flow-with-semantic-kernel
